### ISSUE TITLE
MPS accuracy warning in parse_device_args() is not supported by measurement

### ISSUE THREAD

**Issue by BenjaminDEMAILLE** [scverse/scvi-tools#3980]
### Summary

`parse_device_args()` warns that MPS models "might be slower and **less accurate** than
usual" ([`src/scvi/model/_utils.py:129`](https://github.com/scverse/scvi-tools/blob/main/src/scvi/model/_utils.py#L129)).
I measured this and could not reproduce an accuracy penalty: on the ops that matter for
SVI, MPS float32 is indistinguishable from CPU float32 relative to a float64 reference,
and on the ELBO gradient it was marginally *closer* to the reference.

The warning was added in #2045 (May 2023), when MPS was much less mature, and it points
at [pytorch#77764](https://github.com/pytorch/pytorch/issues/77764), which tracks
**operator coverage**, not numerics. I think the coverage half is still worth warning
about and the accuracy half is no longer supported by evidence.

### Measurements

M4 Max, macOS 26.6.2, torch 2.10.0, pyro 1.9.1. Max absolute error against a CPU
float64 reference, 4096x512 inputs:

| Op | CPU fp32 | MPS fp32 |
|---|---|---|
| `lgamma` | 3.6e-07 | 8.9e-07 |
| `digamma` | 3.9e-06 | 5.4e-06 |
| `log` | 1.8e-07 | 3.3e-07 |
| `exp` | 1.6e-05 | 1.9e-05 |
| `softmax(-1)` | 4.3e-09 | 3.6e-09 |
| `sum(-1)` | 1.8e-04 | 1.7e-04 |
| `matmul` 1024x1024 | 2.07e-06 | 2.07e-06 |

The figure that actually governs training, the gradient of a negative-binomial ELBO
term w.r.t. an AutoNormal location (262k parameters, gradient scale 125):

```
cpu32   max relative deviation from float64   3.8e-07
mps32   max relative deviation from float64   3.7e-07
```

Transcendentals are 1.2x-2.5x further from the reference on MPS, i.e. a ULP or two.
Reductions and matmul are indistinguishable. This is float32 noise on both sides.

<details>
<summary>Reproducer</summary>

```python
import torch, pyro.distributions as dist

def cmp(name, fn, shape=(4096, 512)):
    a64 = torch.rand(*shape, dtype=torch.float64) * 4 + 0.1
    a32 = a64.float()
    ref, cpu = fn(a64), fn(a32).double()
    mps = fn(a32.to("mps")).cpu().double()
    ac = (cpu - ref).abs().max().item()
    am = (mps - ref).abs().max().item()
    print(f"{name:14s} cpu {ac:8.2e} | mps {am:8.2e}")

for n, f in [("lgamma", torch.lgamma), ("digamma", torch.digamma),
             ("log", torch.log), ("exp", torch.exp),
             ("softmax", lambda x: torch.softmax(x, -1)),
             ("sum", lambda x: x.sum(-1))]:
    cmp(n, f)

def grad(dev, dt):
    g = lambda s: torch.Generator().manual_seed(s)
    m = torch.zeros(1024, 256, device=dev, dtype=dt, requires_grad=True)
    s = torch.zeros(1024, 256, device=dev, dtype=dt, requires_grad=True)
    eps = torch.randn(1024, 256, generator=g(1)).to(dev).to(dt)
    r = (torch.rand(1024, 256, generator=g(2)) * 4 + 0.1).to(dev).to(dt)
    cnt = (torch.rand(1024, 256, generator=g(3)) * 8).round().to(dev).to(dt)
    z = eps * s.exp() + m
    dist.GammaPoisson(z.exp(), r).log_prob(cnt).sum().backward()
    return m.grad.cpu().double()

g64, g32, gm = grad("cpu", torch.float64), grad("cpu", torch.float32), grad("mps", torch.float32)
n = g64.abs().max().item()
print("cpu32", (g32 - g64).abs().max().item() / n)
print("mps32", (gm  - g64).abs().max().item() / n)
```

</details>

Two pitfalls I hit first, in case anyone repeats this: relative error against `digamma`
or `lgamma` explodes near those functions' roots on **every** backend, so use absolute
error; and comparing `GammaPoisson.log_prob` on counts produced by `round()` in float32
shows a large discrepancy that comes from .5-boundary rounding of the *input*, not from
the device.

### What I am not claiming

- One machine, one torch version, float32 only. Not a claim about every op or every
  chip.
- RNG streams differ across backends, so MPS and CPU runs will never be bit-identical.
  That is equally true of CUDA vs CPU and is not an accuracy defect.
- Operator coverage remains a real, separate concern. For what it is worth, the gaps
  that mattered to me have been closing: reading the dispatch table in
  `aten/src/ATen/native/native_functions.yaml`, `_standard_gamma` gained an MPS kernel
  in torch 2.12, `_sample_dirichlet` in 2.13, and `poisson` and `binomial` in 2.14.

### Suggested change

Keep the warning, drop the unsupported half:

> `accelerator` has been set to `mps`. Not all PyTorch operations are implemented for
> this backend, so some models may be slower or may fail on unsupported ops; see
> pytorch#77764. Results will not be bit-identical to CPU or CUDA runs because RNG
> streams differ per backend.

Happy to open a PR with that wording, and to add the reproducer under the MPS tests
from #2161 if that is useful.

**Comment by ori-kron-wis** [comment]
@BenjaminDEMAILLE Thank you so much for supporting the MPS/torch integration in scvi-tools and for showing this accuracy stability between devices.
Can you please add this small to the PR you already opened here: https://github.com/scverse/scvi-tools/pull/3981?
